### PR TITLE
Pica: Implement blend factors.

### PR DESCRIPTION
**Pull request by bunnei**
Implements all Pica blend factors except for SourceAlphaSaturate. I wasn't totally sure what that one should look like, so if someone wants to chime in I'll add it to this PR as well.

### CODE REVIEW

**Comment by purpasmart96** [comment]
LGTM, this page might help https://www.opengl.org/sdk/docs/man/html/glBlendFunc.xhtml
or https://www.opengl.org/wiki/GLAPI/glBlendFuncSeparate

File: src/video_core/rasterizer.cpp [hunk 1]
```diff
@@ -528,18 +528,48 @@ void ProcessTriangle(const VertexShader::OutputVertex& v0,
...
+
+                    case params.DestColor:
+                        return dest.rgb();
+
+                    case params.OneMinusDestColor:
+                        return Math::Vec3<u8>(255 - dest.r(), 255 - dest.g(), 255 - dest.b());
+
                     case params.SourceAlpha:
-                        return Math::MakeVec(combiner_output.a(), combiner_output.a(), combiner_output.a());
+                        return Math::Vec3<u8>(combiner_output.a(), combiner_output.a(), combiner_output.a());
```
Comment (neobrain) [inline src/video_core/rasterizer.cpp]: Why?

File: src/video_core/pica.h [hunk 2]
```diff
@@ -357,7 +367,12 @@ struct Regs {
             BitField<0, 4, Op> op;
         } logic_op;
 
-        INSERT_PADDING_WORDS(0x1);
+        union {
+            BitField< 0, 8, u32> r;
+            BitField< 8, 8, u32> g;
+            BitField<16, 8, u32> b;
+            BitField<24, 8, u32> a;
+        } blend_const;
```
Comment (neobrain) [inline src/video_core/pica.h]: Needs to have the two safety checks added at the bottom.

**Comment by neobrain** [comment]
Mostly LGTM

File: src/video_core/rasterizer.cpp (hunk 1 above)
Comment (bunnei) [inline src/video_core/rasterizer.cpp]: Why did this line use MakeVec and the others used Vec3?

**Comment by Alegend45** [comment]
LGTM except for missing SourceAlphaSaturate.

File: src/video_core/pica.h (hunk 2 above)
Comment (bunnei) [inline src/video_core/pica.h]: Why? You already have `output_merger` down there, and you didn't add any of the other member unions of `output_merger`.

Comment (neobrain) [inline src/video_core/pica.h]: Indeed, nevermind.

**Pull request merged**